File: sample-super-rx-ai/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/routing/ListingRouting.java
package com.mateuszkoslacz.moviper.rxsample.viper.routing;

import android.app.Activity;
import android.content.Intent;

import com.mateuszkoslacz.moviper.base.routing.BaseRxRouting;
import com.mateuszkoslacz.moviper.rxsample.viper.contract.ListingContract;
import com.mateuszkoslacz.moviper.rxsample.viper.entity.User;
import com.mateuszkoslacz.moviper.rxsample.viper.presenter.UserDetailsPresenter;
import com.mateuszkoslacz.moviper.rxsample.viper.view.activity.UserDetailsActivity;
import com.mateuszkoslacz.moviper.presentersdispatcher.ActivityStarter;
import com.mateuszkoslacz.moviper.presentersdispatcher.MoviperPresentersDispatcher;

public class ListingRouting
        extends BaseRxRouting<Activity>
        implements ListingContract.Routing {

    @Override
    public void startUserDetailsActivity(User user) {
        if (isContextAttached()) {
            Intent startingIntent = UserDetailsActivity.getStartingIntent(getRelatedContext(), user);
            MoviperPresentersDispatcher.getInstance().startActivity(
                    ActivityStarter.newBuilder()
                            .withContext(getRelatedContext())
                            .withIntent(startingIntent)
                            .withPresenter(new UserDetailsPresenter(startingIntent.getExtras()))
                            .build());
        }
    }
}

File: sample-super-rx-ai-kotlin/src/main/java/com/mateuszkoslacz/moviper/rxsample/viper/routing/ListingRouting.kt
package com.mateuszkoslacz.moviper.rxsample.viper.routing

import android.app.Activity
import com.mateuszkoslacz.moviper.base.routing.BaseRxRouting
import com.mateuszkoslacz.moviper.rxsample.viper.contract.ListingContract
import com.mateuszkoslacz.moviper.rxsample.viper.entity.User
import com.mateuszkoslacz.moviper.rxsample.viper.presenter.UserDetailsPresenter
import com.mateuszkoslacz.moviper.rxsample.viper.view.activity.UserDetailsActivity
import com.mateuszkoslacz.moviper.presentersdispatcher.ActivityStarter
import com.mateuszkoslacz.moviper.presentersdispatcher.MoviperPresentersDispatcher

open class ListingRouting : BaseRxRouting<Activity>(), ListingContract.Routing {

    override fun startUserDetailsActivity(user: User) {
        if (isContextAttached) {
            val startingIntent = UserDetailsActivity.getStartingIntent(relatedContext!!, user)
            MoviperPresentersDispatcher.getInstance().startActivity(
                    ActivityStarter.newBuilder()
                            .withContext(relatedContext!!)
                            .withIntent(startingIntent)
                            .withPresenter(UserDetailsPresenter(startingIntent.extras))
                            .build())
        }
    }
}
